File: decompiled/sources/com/hjq/bar/SelectorDrawable.java
package com.hjq.bar;

import android.graphics.drawable.Drawable;
import android.graphics.drawable.StateListDrawable;

public final class SelectorDrawable extends StateListDrawable {

    public static final class Builder {
        private Drawable mChecked;
        private Drawable mDefault;
        private Drawable mEnabled;
        private Drawable mFocused;
        private Drawable mHovered;
        private Drawable mPressed;
        private Drawable mSelected;

        public Builder setDefault(Drawable drawable) {
            this.mDefault = drawable;
            return this;
        }

        public Builder setFocused(Drawable drawable) {
            this.mFocused = drawable;
            return this;
        }

        public Builder setPressed(Drawable drawable) {
            this.mPressed = drawable;
            return this;
        }

        public Builder setChecked(Drawable drawable) {
            this.mChecked = drawable;
            return this;
        }

        public Builder setEnabled(Drawable drawable) {
            this.mEnabled = drawable;
            return this;
        }

        public Builder setSelected(Drawable drawable) {
            this.mSelected = drawable;
            return this;
        }

        public Builder setHovered(Drawable drawable) {
            this.mHovered = drawable;
            return this;
        }

        public SelectorDrawable build() {
            SelectorDrawable selectorDrawable = new SelectorDrawable();
            Drawable drawable = this.mPressed;
            if (drawable != null) {
                selectorDrawable.addState(new int[]{16842919}, drawable);
            }
            Drawable drawable2 = this.mFocused;
            if (drawable2 != null) {
                selectorDrawable.addState(new int[]{16842908}, drawable2);
            }
            Drawable drawable3 = this.mChecked;
            if (drawable3 != null) {
                selectorDrawable.addState(new int[]{16842912}, drawable3);
            }
            Drawable drawable4 = this.mEnabled;
            if (drawable4 != null) {
                selectorDrawable.addState(new int[]{16842910}, drawable4);
            }
            Drawable drawable5 = this.mSelected;
            if (drawable5 != null) {
                selectorDrawable.addState(new int[]{16842913}, drawable5);
            }
            Drawable drawable6 = this.mHovered;
            if (drawable6 != null) {
                selectorDrawable.addState(new int[]{16843623}, drawable6);
            }
            Drawable drawable7 = this.mDefault;
            if (drawable7 != null) {
                selectorDrawable.addState(new int[0], drawable7);
            }
            return selectorDrawable;
        }
    }
}

File: decompiled/sources/com/hjq/bar/SelectorDrawable.kt
package com.hjq.bar

import android.graphics.drawable.Drawable
import android.graphics.drawable.StateListDrawable

class SelectorDrawable : StateListDrawable() {
    class Builder {
        private var mChecked: Drawable? = null
        private var mDefault: Drawable? = null
        private var mEnabled: Drawable? = null
        private var mFocused: Drawable? = null
        private var mHovered: Drawable? = null
        private var mPressed: Drawable? = null
        private var mSelected: Drawable? = null

        fun setDefault(drawable: Drawable?): Builder {
            mDefault = drawable
            return this
        }

        fun setFocused(drawable: Drawable?): Builder {
            mFocused = drawable
            return this
        }

        fun setPressed(drawable: Drawable?): Builder {
            mPressed = drawable
            return this
        }

        fun setChecked(drawable: Drawable?): Builder {
            mChecked = drawable
            return this
        }

        fun setEnabled(drawable: Drawable?): Builder {
            mEnabled = drawable
            return this
        }

        fun setSelected(drawable: Drawable?): Builder {
            mSelected = drawable
            return this
        }

        fun setHovered(drawable: Drawable?): Builder {
            mHovered = drawable
            return this
        }

        fun build(): SelectorDrawable {
            val selectorDrawable = SelectorDrawable()
            mPressed?.let { selectorDrawable.addState(intArrayOf(16842919), it) }
            mFocused?.let { selectorDrawable.addState(intArrayOf(16842908), it) }
            mChecked?.let { selectorDrawable.addState(intArrayOf(16842912), it) }
            mEnabled?.let { selectorDrawable.addState(intArrayOf(16842910), it) }
            mSelected?.let { selectorDrawable.addState(intArrayOf(16842913), it) }
            mHovered?.let { selectorDrawable.addState(intArrayOf(16843623), it) }
            mDefault?.let { selectorDrawable.addState(IntArray(0), it) }
            return selectorDrawable
        }
    }
}
